src/providers/twitch/api/Helix.cpp

```diff
@@ -922,6 +922,98 @@ void Helix::deleteChatMessages(
         .execute();
 }
 
+void Helix::addChannelModerator(
+    QString broadcasterID, QString userID, ResultCallback<> successCallback,
+    FailureCallback<HelixAddChannelModeratorError, QString> failureCallback)
+{
+    using Error = HelixAddChannelModeratorError;
+
+    QUrlQuery urlQuery;
+
+    urlQuery.addQueryItem("broadcaster_id", broadcasterID);
+    urlQuery.addQueryItem("user_id", userID);
+
+    this->makeRequest("moderation/moderators", urlQuery)
+        .type(NetworkRequestType::Post)
+        .onSuccess([successCallback, failureCallback](auto result) -> Outcome {
+            if (result.status() != 204)
+            {
+                qCWarning(chatterinoTwitch)
+                    << "Success result for deleting chat messages was"
+                    << result.status() << "but we only expected it to be 204";
+            }
+
+            successCallback();
+            return Success;
+        })
+        .onError([failureCallback](auto result) {
+            auto obj = result.parseJson();
+            auto message = obj.value("message").toString();
+
+            switch (result.status())
+            {
+                case 403: {
+                    // 403 endpoint means the user does not have permission to perform this action in that channel
+                    // Most likely to missing moderator permissions
+                    // Missing documentation issue: https://github.com/twitchdev/issues/issues/659
+                    // `message` value is well-formed so no need for a specific error type
+                    failureCallback(Error::Forwarded, message);
+                }
+                break;
+
+                case 401: {
+                    if (message.startsWith("Missing scope",
+                                           Qt::CaseInsensitive))
+                    {
+                        // Handle this error specifically because its API error is especially unfriendly
+                        failureCallback(Error::UserMissingScope, message);
+                    }
+                    else
+                    {
+                        failureCallback(Error::Forwarded, message);
```

Comment:
```json
{
  "error": "Unauthorized",
  "message": "incorrect user authorization",
  "status": 401
}
```

This message is spit out if the user is trying to mod someone in the wrong channel, let's catch this and make it a friendlier error

Reply:
done